src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineIntegrationTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Tests\Transport\Doctrine;
+
+use Doctrine\DBAL\DriverManager;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+use Symfony\Component\Messenger\Transport\Doctrine\Connection;
+
+/**
+ * @requires pdo_mysql
+ */
+class DoctrineIntegrationTest extends TestCase
+{
+    private $driverConnection;
+    private $connection;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (!getenv('MESSENGER_DOCTRINE_DSN')) {
+            $this->markTestSkipped('The "MESSENGER_DOCTRINE_DSN" environment variable is required.');
+        }
+        $dsn = getenv('MESSENGER_DOCTRINE_DSN');
+        $this->driverConnection = DriverManager::getConnection(['url' => $dsn]);
+        $this->connection = new Connection([], $this->driverConnection);
+        // call publish to auto-setup the table
+        $this->connection->publish('...', []);
+        // ensure the table is clean for tests
+        if (false !== strpos($dsn, 'sqlite')) {
+            $this->driverConnection->exec('DELETE FROM messenger_messages');
+        } else {
+            $this->driverConnection->exec('TRUNCATE messenger_messages');
+        }
+
+        // setup the table
+        $reflection = new \ReflectionClass($this->connection);
+        $setup = $reflection->getMethod('setup');
+        $setup->setAccessible(true);
+        $setup->invoke($this->connection);
+    }
+
+    public function testConnectionPublishAndGet()
+    {
+        $this->connection->publish('{"message": "Hi"}', ['type' => DummyMessage::class]);
+        $encoded = $this->connection->get();
+        $this->assertEquals('{"message": "Hi"}', $encoded['body']);
+        $this->assertEquals(['type' => DummyMessage::class], $encoded['headers']);
+    }
+
+    public function testPublishWithDelay()
+    {
+        $now = \DateTime::createFromFormat('U.u', microtime(true));
+        $this->connection->publish('{"message": "Hi i am delayed"}', ['type' => DummyMessage::class], 600000);
+
+        $available_at = $this->driverConnection->createQueryBuilder()
+            ->select('m.available_at')
+            ->from('messenger_messages', 'm')
+            ->where('m.body = :body')
+            ->setParameter(':body', '{"message": "Hi i am delayed"}')
+            ->execute()
+            ->fetchColumn();
+
+        $available_at = new \DateTime($available_at);
+
+        $this->assertGreaterThanOrEqual($now, $available_at);
```

Comment:
Would `assertGreaterThan()` be better? Equal could actually happen if we forgot to use the `600000` value.